### PR TITLE
[TRAFODION-2604] sort merge phase memory pool improvements

### PR DESCRIPTION
**Pull request by prashanth-vasudev**

### CODE REVIEW

File: core/sql/executor/ex_sort.cpp [hunk 1]
```diff
@@ -218,7 +304,8 @@ ExSortTcb::ExSortTcb(const ExSortTdb & sort_tdb,
   sortStats_ = NULL;
   childTcb_ = &child_tcb;
 
-  CollHeap * space = glob->getSpace();
+  //Create heap to be used by sort.
+  sortHeap_ = new(getHeap()) NAHeap("Sort Heap", (NAHeap *)getHeap(), 204800);
```
Comment (DaveBirdsall) [inline core/sql/executor/ex_sort.cpp]: What's the reasoning behind the hard-coded size, 204800?

Comment (sandhyasun) [inline core/sql/executor/ex_sort.cpp]: Isn't there a SORT_MAX_HEAP_SIZE_MB CQD ?

**Comment by sandhyasun** [comment]
+1 Looks good to me too.

**Comment by selvaganesang** [comment]
Please attempt to use NADELETE instead of delete.

File: core/sql/executor/ex_sort.cpp (hunk 1 above)
Comment (prashanth-vasudev) [inline core/sql/executor/ex_sort.cpp]: SortHeap is used to allocate internal data structures to accommodate initial default number of 1 million records. The hard coded value of 200kb is the block size hint given to NAheap. This value will help allocations that are not too big or too small for sort.  SortHeap is initialized only at one place in sort and also this value is not a tune-able parameter , hence kept it as is and did not make it as a define or cqd. 

SORT_MAX_HEAP_SIZE_MB cqd defaults to 800MB , this is the max limit that will be enforced if quota system is not enforced for sort.

**Pull request merged**